src/vcard/mod_vcard_api.erl

```diff
@@ -0,0 +1,281 @@
+%methods used in vcard graphQL API
+-module(mod_vcard_api).
+
+-include("mongoose.hrl").
+-include("jlib.hrl").
+-include("mod_vcard.hrl").
+
+-export([transform_from_graphql/1,
+         set_vcard/2,
+         to_gql_format/1,
+         get_vcard/1,
+         name_components_proccess/2,
+         address_components_proccess/2,
+         label_components_proccess/2,
+         telephone_components_proccess/2,
+         email_components_proccess/2,
+         geo_components_proccess/2,
+         org_components_proccess/2
+        ]).
+
+-ignore_xref([name_components_proccess/2,
+              address_components_proccess/2,
+              label_components_proccess/2,
+              telephone_components_proccess/2,
+              email_components_proccess/2,
+              geo_components_proccess/2,
+              transform_from_graphql/1,
+              to_gql_format/1,
+              org_components_proccess/2]).
+
+set_vcard(#jid{luser = LUser, lserver = LServer} = UserJID, Vcard) ->
+    case mongoose_domain_api:get_domain_host_type(LServer) of
+        {ok, HostType} ->
+            case set_vcard(HostType, UserJID, Vcard) of
+                ok ->
+                    case get_vcard(HostType, LUser, LServer) of
+                        {ok, _} = Result ->
+                            Result;
+                        Error ->
+                            Error
+                    end;
+                Error ->
+                    Error
+            end;
+        Error ->
+            Error
+    end.
+
+get_vcard(#jid{luser = LUser, lserver = LServer}) ->
+    case mongoose_domain_api:get_domain_host_type(LServer) of
+        {ok, HostType} ->
+            case get_vcard(HostType, LUser, LServer) of
+                {ok, _} = Vcard ->
+                    Vcard;
+                Error ->
+                    Error
+            end;
+        Error ->
+            Error
+    end.
+
+set_vcard(HostType, UserJID, Vcard) ->
+    mod_vcard:unsafe_set_vcard(HostType, UserJID, transform_from_graphql(Vcard)).
```

Comment:
What is the reason why `set` uses the unsafe variant while `get` has a `try...catch`? It looks inconsistent. maybe we don't need to be so defensive in `get_vcard`? @Premwoik how does this compare to the other modules?